types/BUILD.bazel

```diff
@@ -28,6 +28,9 @@ js_binary(
         ":types_lib",
     ],
     entry_point = "scripts/build-types.js",
+    env = {
+        "NODE_COMPILE_CACHE": "/tmp/workerd-types-compile-cache",
```

Comment:
@fhanau @mikea any suggestions on what to pass here?

Reply:
I mean... do we need this? This action is only used to generate types right?

Reply:
Not strictly needed, no, but results in a significant performance improvement when used correctly.

Reply:
It can be any temp directory, but something that can be preserved in remote cache as well would be really good.

Reply:
you'd need
- add data dependency to the target
- use `$(location)` to set env var